core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -138,6 +148,49 @@ oppia.factory('SimpleEditorManagerService', [
         data.questionList.addQuestion(QuestionObjectFactory.create(
           lastStateName, stateData.interaction, ''));
       },
+      deleteQuestion: function(question){
+        // - Change destination of answer groups that point to it.
+        // - Move content stored in present state to next state if it exists.
+        // - Delete the state.
+        var stateName = question.getStateName();
+        var state = SimpleEditorShimService.getState(stateName);
+        // If it's the last question in the list, make it EndExploration.
+        if(state.interaction.answerGroups.length == 0){
+          makeStateEndExploration(stateName);
+          data.questionList.removeQuestion(question);
+          return;
+        }
+        var nextStateName = state.interaction.answerGroups[0].outcome.dest;
+        var allStateNames = SimpleEditorShimService.getAllStateNames();
+        // Change init state name, if init_state is being deleted.
+        if(SimpleEditorShimService.getInitStateName() == stateName){
+          explorationInitStateNameService.displayed = nextStateName;
+          explorationInitStateNameService.saveDisplayedValue(nextStateName);
+        }
+
+        for(var i = 0; i < allStateNames.length;i++){
+          var currentState = SimpleEditorShimService.getState(allStateNames[i]);
+          var newAnswerGroups = currentState.interaction.answerGroups;
+          var hasAnswerGroupsChanged = false;
+          for(var j = 0; j < currentState.interaction.answerGroups.length;j++){
```

Comment:
I am using 'j' because I have two copies of data, one is newAnswerGroups - modifications are done in this while a second copy is in currentState.interaction.answerGroups[i], if I remove a copy and change things in original (i.e. currentState.interaction.answerGroups[i]) then I also I have to use 'j' because forEach will send a copy one by one ( am I right?) and It won't be possible to retain them after the forEach ends. In case I keep the two copies as it is then I need 'j' to refer to same j'th answerGroup in both the copies.